contracts/src/v0.8/automation/dev/v2_3/AutomationRegistrar2_3.sol

```diff
@@ -131,7 +131,8 @@ contract AutomationRegistrar2_3 is TypeAndVersionInterface, ConfirmedOwner, IERC
     bytes triggerConfig,
     bytes offchainConfig,
     bytes checkData,
-    uint96 amount
+    uint96 amount,
+    IERC20 billingToken
```

Comment:
since we have to change this event anyway.. I think we could better "future proof" it by changing it to just include the whole `RegistrationParams` struct. wdyt?
```solidity
event RegistrationRequested(
    bytes32 indexed hash,
    RegistrationParams registrationParams
)
```

Reply:
Using registrationParams is far neater, I like it! Pasting it here for reference:

```
  struct RegistrationParams {
    address upkeepContract;
    uint96 amount;
    // 1 word full
    address adminAddress;
    uint32 gasLimit;
    uint8 triggerType;
    // 7 bytes left in 2nd word
    IERC20 billingToken;
    // 12 bytes left in 3rd word
    string name;
    bytes encryptedEmail;
    bytes checkData;
    bytes triggerConfig;
    bytes offchainConfig;
  }
```

I confirmed registrationParams contains exactly the same the params needed, except hash, which we will add.

For an Event (RegistrationRequested) which has a Struct (RegistrationParams), the hash will be just listing all the fields in the struct, so keccak(bytes32,address upkeepContract,uint96 amount,address....)?

I believe that is how it works, but double confirm @RyanRHall ?
We will need to function signature in atlas, hence confirming, but I dont think using the RegistrationParams will require any other changes except the hash. cc @cmalec

Reply:
ya for new versions of the event we will just need to decode RegistrationParams but should be fine.

Reply:
Note to myself, the event signature is now:

- RegistrationRequested(bytes32,address,uint96,address,uint32,uint8,address,string,bytes,bytes,bytes,bytes)
- 0x72319f308e224802c7dc5f4bad21a9b46070f765d25eedf39a25f04bf39eb632

Reply:
I realized that we introduced some side effects bc of this change. Given that we dont change the params in the event often, I reverted this change. More details in the other comment.